### ISSUE TITLE
Change the configuration of the project

### ISSUE THREAD

**Issue by jhonalex06** [opengisch/qfieldsync#117]
[Short description of problem here]

When we use ModelBaker to generate a new project with relations between the differents tables or layers, these relations are preserved inside the QGIS project. However, when we use QFieldSync to package the project for QField the project loses the relations and the layer properties. This happens when we use Geopackage data.

### Reproduction Steps:

1. Open QGIS
2. Open the menu Generate of ModelBaker, and select Interlis (use GeoPackage).
3. In the Interlis File, select the file: Captura_Geografica_V0_3.ili, this shows you a model: Captura_Geografica_V0_3
4. Select the path of the geopackage and click in Create.
5. When finish, save the project.
6. Now, Use QFieldSync to configure the project, select all layer editing offline.
7. Select the folder where you want to save the project and click in create.
8. Now, you have two projects for making tests, the project generated in the step 5 and the project generated in the step 7.
9. Open the project in the step 5 inside QField and create some features with relations. (You won’t see any problem).
10 Now Open the project generated in the step 7 with QFieldSync and try to create features with relations. In that moment you will see the error: Cannot add child. Parent primary keys are not available.
 
### Expected behavior:

We expected that the configuration in the project will be maintained when we used QFieldSync.

### Observed behavior:

The configuration in the projecto not be maintained when we used QFieldSync.

### Screenshots and GIFs

![issue](https://user-images.githubusercontent.com/27906888/59537993-ed237d80-8ebd-11e9-8180-6bddf5fd49fb.png)

[Please also attach additional files if a specific project/dataset is useful to investigate the problem.]

Archivos: ili, project generated with QFieldSync, project generated without QFieldSync.

**Qgis ModelBaker version:** 4.2.2
**QFieldSync version:** 3.1.3
**QField version:** 1.0.3 - Matterhorn (10003990)

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: No
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: Yes

**Comment by m-kuhn** [comment]
**Our recommendation for the current moment when using relations is not to use QFieldSync and directly prepare a QGIS project using UUIDs.**

# Reasoning

It would indeed be very nice to preserve this configuration when running a project through QFieldSync.
In case of directly copying files (or direct connections to the database) this should work fairly well.
The offline editing case is a bit more tricky though. In particular, if several offline copies are checked out in parallel, it needs to be guaranteed that after integration into the database, the there are no conflicts and features still match. There are different strategies for this:

 * Generate ids that are guaranteed to be unique (e.g. UUID) on the field already
   * Requirement: database needs to be setup accordingly
   * QFieldSync needs to be aware of this and configure the project accordingly
 * Regenerate ids while merging with the database
   * QFieldSync also needs to generate a default value expression (autoconfigured?)
   * QFieldSync needs to be aware of the dependencies and merge parents before children and cache the newly generated ids

To summarize, it's a desired feature but not trivial and there is no "one size suits all".
If you have a project where you would like that implemented, please contact us directly so we can work on specifications that will work for your specific use case.

**Comment by jhonalex06** [comment]
Solved https://github.com/opengisch/QField/pull/689
